containers/src/Kokkos_DualView.hpp

```diff
@@ -832,16 +836,17 @@ void
 deep_copy (DualView<DT,DL,DD,DM> dst, // trust me, this must not be a reference
            const DualView<ST,SL,SD,SM>& src )
 {
-  if(src.modified_flags.data()==NULL || dst.modified_flags.data()==NULL) {
+  if(src.modified_flags_data()==NULL || dst.modified_flags_data()==NULL) {
     return deep_copy(dst.d_view, src.d_view);
   }
-  if (src.modified_flags(1) >= src.modified_flags(0)) {
-    deep_copy (dst.d_view, src.d_view);
-    dst.template modify<typename DualView<DT,DL,DD,DM>::device_type> ();
-  } else {
+  if ( src.need_sync_device() ) {
```

Comment:
A couple preemptive comments:
1.  I changed the check from `src.modified_flags(1) >= src.modified_flags(0)` to `src.modified_flags(1) < src.modified_flags(0)` (and swapped the if-else contents) to replace `src.modified_flags(1) < src.modified_flags(0)` with `src.need_sync_device()`
2. The `src.need_sync_device()` will return false if data is null, however this will get caught in line 839 above.